src/fshtypes/Model.ts

```diff
@@ -0,0 +1,20 @@
+import { FshEntity } from './FshEntity';
+import { SdRule, AddElementRule } from './rules';
+
+export abstract class Model extends FshEntity {
```

Comment:
This is very close to the same definition as `Profile` (and only one more degree different than `Extension`).  Neither of those have a base class other than `Entity`.  Do you think it might make sense to have them all share a base class that has `Rules: Rule[]` and override that in the sub-classes?

I'm not sure if `Model` is quite the right word (whether shared among all four or just these two) -- but I'm struggling to find a better one...  I guess `Structure` might be more accurate in context, but maybe would be confusing due to the existence of a `StructureDefinition` class as well?  I guess my concern w/ `model` is that people might associate it exclusively with Logical Models.

Reply:
Ok @cmoesel, https://github.com/FHIR/sushi/pull/784/commits/0235b08b4c970a81125832112f9e2419a08b9b21 adds a `FshStructure` class which `Profile` and `Extension` now also inherit from.